pkg/asset/agent/manifests/infraenv.go

```diff
@@ -131,5 +145,9 @@ func (i *InfraEnv) finish() error {
 		return errors.New("missing configuration or manifest file")
 	}
 
+	// Throw an error if CpuArchitecture isn't RpmArch
+	if i.Config.Spec.CpuArchitecture != arch.RpmArch(i.Config.Spec.CpuArchitecture) {
```

Comment:
Given the [implementation](https://github.com/openshift/installer/blob/master/vendor/github.com/coreos/stream-metadata-go/arch/arch.go#L31-L38) of `RpmArch()`, I don't think this is a sufficient check. e.g. the hitherto-unknown CPU architecture `"fred"` would not cause an error here. In fact, there is no possible way to trigger an error here, given that we always do this conversion in `Load()`.

Reply:
@zaneb I've updated/pushed the logic here: https://github.com/openshift/installer/pull/6401/files#diff-2d7fa80a467b94ef8ec0dd77511eeeb42630c9d1ccea2c5e21eaf041d6f1ed73R147